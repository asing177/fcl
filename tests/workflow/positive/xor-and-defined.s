global int x;

@initial
split()
{
  transitionTo(@{a1,b});
}

@a1
init1()
{
  x = 0;
  transitionTo(@a2);
}

@{a2,b}
join()
{
  transitionTo(@commonState);
}

@initial
init2()
{
  x = 0;
  transitionTo(@commonState);
}

@commonState
use()
{
  y = x;
  terminate();
}
