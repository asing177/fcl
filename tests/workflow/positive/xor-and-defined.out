 
Method arguments and effects:
split: () -> {}
init1: () -> {write x}
join: () -> {}
init2: () -> {write x}
use: () -> {write via terminate, read x}